lib/puppet/functions.rb

```diff
@@ -108,6 +116,22 @@
 #     end
 #   end
 #
+# There is no requirement for direct mapping between parameter declarations and the parameters in the receiver so
```

Comment:
definitions instead of declarations. "receiver" is also referred to as "imlementation method" (maybe explain that separately).